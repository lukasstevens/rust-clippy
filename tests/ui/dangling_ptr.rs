#![warn(dangling_ptr)]

#[allow(unused_variables)]
fn main() {
    let x = 1 as *const usize;
    let y = 1 as *mut f64;

    let z = 1;
    let z = z as *const usize; // this is currently not caught
}
