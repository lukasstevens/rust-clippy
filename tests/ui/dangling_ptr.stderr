error: `1 as *const _` detected. Consider using `ptr::dangling()`
 --> $DIR/dangling_ptr.rs:5:13
  |
5 |     let x = 1 as *const usize;
  |             ^^^^^^^^^^^^^^^^^
  |
  = note: `-D dangling-ptr` implied by `-D warnings`

error: `1 as *mut _` detected. Consider using `ptr::dangling_mut()`
 --> $DIR/dangling_ptr.rs:6:13
  |
6 |     let y = 1 as *mut f64;
  |             ^^^^^^^^^^^^^

